RBAC: Allow listing user permissions with scope (#57538)
grafana/grafana: docs/sources/administration/service-accounts/index.md

1. In the **Permissions** section at the bottom, click **Add permission**.
1. Choose **User** in the dropdown and select your desired user.
1. Choose **View**, **Edit** or **Admin** role in the dropdown and click **Save**.

## Debug the permissions of a service account token

> Sometimes, for debugging purposes, you might need to know which RBAC permissions are attached to a service account token.
> This section will give you an overview of how to do it.

Reviewer: ```suggestion
This section explains how to learn which RBAC permissions are attached to a service account token.
This can help you diagnose permissions-related issues with token authorization.
```

Author: perfect!

### Before you begin

> Make sure to read the section to [add a token to a service account]({{< relref "#add-a-token-to-a-service-account-in-grafana" >}})
> as we will need to use the token to authenticate against Grafana.

Reviewer: ```suggestion
These endpoints provide details on a service account's token.
If you haven't added a token to a service account, do so before proceeding.
For details, refer to [Add a token to a service account]({{< relref "#add-a-token-to-a-service-account-in-grafana" >}}).
```

### List a service account token's permissions

> To list your token's permissions, you can target the `/api/access-control/user/permissions` permissions.

Reviewer: ```suggestion
To list your token's permissions, use the `/api/access-control/user/permissions` endpoint.
```

#### Example

> > note that the following command output is shortened

Reviewer: ```suggestion
> **Note:** The following command output is shortened to show only the relevant content.
```

> 

Reviewer: ```suggestion
Authorize your request with the token whose permissions you want to check.

```

```bash

> curl -H "Authorization: Bearer glsa_HOruNAb7SOiCdshU9algkrq7FDsNSLAa_54e2f8be" -X GET '<grafana_url>/api/access-control/user/permissions?scoped=true' | jq

Reviewer: No action necessary in this PR, just noting: This is a token authorization example, and we could also use one on https://grafana.com/docs/grafana/latest/developers/http_api/curl-examples/, which right now only has basic auth examples.

CC @GrafanaWriter for dev docs consideration

```

> output:

Reviewer: ```suggestion
The output lists the token's permissions:
```

```json
{
  "dashboards:read": ["dashboards:uid:70KrY6IVz"],
  "dashboards:write": ["dashboards:uid:70KrY6IVz"],
  "datasources.id:read": ["datasources:*"],
  "datasources:read": ["datasources:*"],
  "datasources:explore": [""],
  "datasources:query": ["datasources:uid:grafana"],
  "datasources:read": ["datasources:uid:grafana"],
  "orgs:read": [""]
}
```

### Check which dashboards a token is allowed to see

> To see which dashboards a token is allowed to see, you can target the `/api/access-control/user/permissions` permissions.

Reviewer: ```suggestion
To list which dashboards a token can view, you can filter the `/api/access-control/user/permissions` endpoint's response for the `dashboards:read` permission key.
```

#### Example

```bash

> curl -H "Authorization: Bearer glsa_HOruNAb7SOiCdshU9algkrq7FDsNSLAa_54e2f8be" -X GET '<grafana_url>/api/access-control/user/permissions?scoped=true' | jq '."dashboards:read"'

Reviewer: Is `scoped` required to filter on dashboard permissions?

Author: Yes to be able to see the scope list, otherwise the user would only see the action.
But @kalleep is challenging using the same endpoint all together so I'm changing this.

```

> output:

Reviewer: ```suggestion
The output lists the token's viewable dashboards by their unique identifiers (`uid`):
```

#### Example

> **Note:** The following command output is shortened to show only the relevant content.
> Authorize your request with the token whose permissions you want to check.

```bash
curl -H "Authorization: Bearer glsa_HOruNAb7SOiCdshU9algkrq7FDsNSLAa_54e2f8be" -X GET '<grafana_url>/api/access-control/user/permissions' | jq
```

The output lists the token's permissions:

```json
{
  "dashboards:read": ["dashboards:uid:70KrY6IVz"],
  "dashboards:write": ["dashboards:uid:70KrY6IVz"],
  "datasources.id:read": ["datasources:*"],
  "datasources:read": ["datasources:*"],
  "datasources:explore": [""],
  "datasources:query": ["datasources:uid:grafana"],
  "datasources:read": ["datasources:uid:grafana"],
  "orgs:read": [""]
}
```

### Check which dashboards a token is allowed to see

To list which dashboards a token can view, you can filter the `/api/access-control/user/permissions` endpoint's response for the `dashboards:read` permission key.

#### Example

```bash
curl -H "Authorization: Bearer glsa_HOruNAb7SOiCdshU9algkrq7FDsNSLAa_54e2f8be" -X GET '<grafana_url>/api/access-control/user/permissions' | jq '."dashboards:read"'
```

> The output lists the token's viewable dashboards by their unique identifiers (`uid`):

Reviewer 2: nit: it could also contain UIDs of folders, and user would be able to read all the dashboards in that folder. I think it should be mentioned to avoid potential confusion.

Author: Done
